File: app/src/main/java/com/hellmund/primetime/utils/NotificationPublisher.java
package com.hellmund.primetime.utils;

import android.app.Notification;
import android.app.NotificationManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.hellmund.primetime.model.WatchlistMovie;

import java.util.ArrayList;
import java.util.Calendar;

public class NotificationPublisher extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (isCorrectTime() && NotificationUtils.areNotificationsEnabled(context)) {
            ArrayList<WatchlistMovie> releases = new ArrayList<>(); // Watchlist.getReleasesToday();
            if (releases.isEmpty()) {
                return;
            }

            NotificationManager notificationMgr =
                    (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
            Notification notification = NotificationUtils.buildNotification(context, releases);
            notificationMgr.notify(0, notification);
        }
    }

    private boolean isCorrectTime() {
        Calendar now = Calendar.getInstance();
        final int hour = now.get(Calendar.HOUR_OF_DAY);
        return hour == 9;
    }

}

File: app/src/main/java/com/hellmund/primetime/utils/NotificationPublisher.kt
package com.hellmund.primetime.utils

import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.core.content.getSystemService
import com.hellmund.primetime.di.app
import com.hellmund.primetime.ui.watchlist.WatchlistRepository
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import org.threeten.bp.LocalDate
import javax.inject.Inject

class NotificationPublisher : BroadcastReceiver() {

    @Inject
    lateinit var watchlistRepository: WatchlistRepository

    override fun onReceive(context: Context, intent: Intent) {
        context.app.appComponent.inject(this)

        if (!NotificationUtils.areNotificationsEnabled(context)) {
            return
        }

        GlobalScope.launch {
            val releases = watchlistRepository.getReleases(LocalDate.now())
            if (releases.isEmpty()) {
                return@launch
            }

            val notification = NotificationUtils.buildNotification(context, releases)
            val notificationManager = context.getSystemService<NotificationManager>()
            notificationManager?.notify(0, notification)
        }
    }

}
